driver/terraform.go

```diff
@@ -487,6 +487,15 @@ func (tf *Terraform) initTaskTemplate() error {
 		Renderer:     renderer,
 		FuncMapMerge: tmplfunc.HCLMap(metaMap),
 	})
+
+	if tf.template != nil && tf.template.ID() == tmpl.ID() {
```

Comment:
It may be the case where the existing template lingers indefinitely when the new template content differs for this task, wrt to the hcat.Watcher.

Is there a way to clear the older, un-used template from the Watcher? Is [`Watcher.Sweep`](https://pkg.go.dev/github.com/hashicorp/hcat#Watcher.Sweep) appropriate here when `tf.template != nil && tf.template.ID() != tmpl.ID()`

Reply:
Thanks for the feedback! Yes, it makes sense to add a cleanup! I went ahead and added it.

Noting also that we chatted that there are other issues with this PR to work through to resolve the original issue.